app.py

```diff
@@ -28,53 +28,43 @@ def get_conversation_thread(client, channel, ts):
         user_id = message["user"]
         if user_id not in users:
             user_info = client.users_info(user=user_id)
+            # print(user_info["user"])
             users[user_id] = {
-                "name": user_info["user"]["profile"]["real_name_normalized"],
-                "profile_picture": user_info["user"]["profile"]["image_24"]
+                "real_name": user_info["user"]["real_name"],
             }
 
         messages.append({
             "text": message["text"],
-            "name": users[user_id]["name"],
-            "profile_picture": users[user_id]["profile_picture"]
+            "name": users[user_id]["real_name"],
         })
 
         if "files" in message:
             permalinksToAttachments = ""
             for file in message["files"]:
                 if not file['public_url_shared']:
                     app.client.files_sharedPublicURL(file=file["id"], token=os.environ.get("USER_TOKEN"))
-                permalinksToAttachments = f'{permalinksToAttachments}<a href="{file["permalink_public"]}">Download</a><p>'
+                if file["filetype"]=="png" or file["filetype"]=="jpg" or file["filetype"]=="jpeg": #file is image
+                    permalinksToAttachments = f'{permalinksToAttachments}\n<br>![Download]({file["url_private_download"]}?pub_secret={file["permalink_public"][-10:]})'
```

Comment:
hmm so I'm not sure if file["permalink_public][-10:] is what we want to do, because what if the key is not ten characters long?
I think the right way to do this is to split on the "=", and then take the last segment, which would be something like file["permalink_public"].split("=")[-1]